.github/workflows/R-CMD-check.yaml

```diff
@@ -104,6 +104,18 @@ jobs:
         run: rcmdcheck::rcmdcheck(args = c("--no-manual", "--as-cran"), error_on = "warning", check_dir = "check")
         shell: Rscript {0}
 
+      - name: Install cmdstan, stan, math from develop
+        run: |
+          cmdstanr::install_cmdstan(cores = 2, cmdstan_branch = "develop",
+                                    stan_branch = "develop", math_branch = "develop")
+        shell: Rscript {0}
+
+      - name: Check against upstream development
+        env:
+          _R_CHECK_CRAN_INCOMING_: false
+        run: rcmdcheck::rcmdcheck(args = c("--no-manual", "--as-cran"), error_on = "warning", check_dir = "check")
```

Comment:
This would run these tests in each of the test runs. I think that is a bit overkill. These tests take about an hour as is already. My idea would be to make a separate run that would run the full tests with the `develop` version of cmdstan on Linux with the release version of R. That would be helpful for multiple reasons.